packages/expect-utils/src/jasmineUtils.ts

```diff
@@ -83,7 +83,9 @@ function eq(
   }
 
   if (a instanceof Error && b instanceof Error) {
-    return a.message == b.message;
+    // Compare just error messages only if not strict check
+    if (!strictCheck) return a.message == b.message;
+    if (a.message != b.message) return false;
```

Comment:
this seems somewhat odd to me - should we stop comparing Errors unless their message matches? Is the thought that we want to treat that as any other objects?

Reply:
With the previous implementation, the discriminant determining `Error`s equality was only the `message` field.
With this implementation, you compare only the messages for non-strict equality, and you treat the two objects as any other object for strict equality checks.

It probably depends on how you think strict equality should be intended for `Error` objects.